File: app/src/main/java/com/samet/coinApp/CoinApplication.java
package com.samet.coinApp;

import com.samet.coinApp.di.components.AppComponent;
import com.samet.coinApp.di.components.DaggerAppComponent;
import com.samet.coinApp.di.modules.ApiClientModule;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

/**
 * Created by Samet Albayrak on 24/04/19.
 */
public class CoinApplication extends DaggerApplication {

    @Override
    public void onCreate() {
        super.onCreate();
    }


    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = getAppComponent();
        appComponent.inject(this);
        return appComponent;
    }

    private AppComponent getAppComponent() {
        ApiClientModule apiClientModule = new ApiClientModule(this);
        return DaggerAppComponent.builder()
                .apiClientModule(apiClientModule)
                .application(this).build();

    }


}

File: app/src/main/java/com/samet/coinApp/CoinApplication.kt
package com.samet.coinApp

import android.app.Application
import com.onesignal.OneSignal
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class CoinApplication : Application(){

    override fun onCreate() {
        super.onCreate()

        // OneSignal Initialization
        OneSignal.startInit(this)
            .inFocusDisplaying(OneSignal.OSInFocusDisplayOption.Notification)
            .unsubscribeWhenNotificationsAreDisabled(true)
            .init()

        // Admob Initialization
        // MobileAds.initialize(this, getString(R.string.admob_app_id))

    }

}